### PR TITLE
Fixed Issues related to dynamic dashboard, gadget maximization and banners

### PR DESCRIPTION
**Pull request by lasanthaS**

### CODE REVIEW

File: apps/portal/js/dashboard.js [hunk 1]
```diff
@@ -12,61 +12,53 @@ $(function () {
...
-                renderComponentMaxView(uesComponent);
             }
         });
 
-        $('#wrapper').on('click', 'a.ues-component-settings-handle', function () {
-            var id = $(this).closest('.ues-component').attr('id');
-            var component = findComponent(id);
-            componentContainer = $('#'+containerPrefix + id);
+        $('#wrapper').on('click', 'a.ues-component-settings-handle', function (e) {
+
```
Comment (TanyaM) [inline apps/portal/js/dashboard.js]: New line

File: apps/portal/js/dashboard.js [hunk 2]
```diff
@@ -106,41 +103,35 @@ $(function () {
...
      */
-    var renderComponentToolbar = function (component) {        
-        var el = $('#' + component.id).prepend($(componentToolbarHbs(component.content)));
-        // hide the settings button from the anon view
-        if (ues.global.dbType === 'anon') {
-            $('a.ues-component-settings-handle', el).hide();
+    var renderComponentToolbar = function (component) {
+        
+        if (component) {
+
```
Comment (TanyaM) [inline apps/portal/js/dashboard.js]: New line

File: apps/portal/js/designer.js [hunk 3]
```diff
@@ -597,32 +593,51 @@ $(function () {
...
+            
+            var componentBox = $(this).closest('.ues-component-box'), 
+                id = componentBox.find('.ues-component').attr('id'), 
+                removeBlock = true;
+            
+            if (id) {
+                removeComponent(findComponent(id), function (err) {
+                    if (err) {
+                        removeBlock = false;
+                        console.error(err);
```
Comment (TanyaM) [inline apps/portal/js/designer.js]: if console is not present

File: apps/portal/js/designer.js [hunk 4]
```diff
@@ -1791,36 +1758,26 @@ $(function () {
...
-                        if (!hasComponents($(this))) {
-                            moveComponent(el, id);
-                        }
-                        break;
-                    default:
-                        if (!hasComponents($(this))) {
-                            createComponent(el, findStoreCache(type, id));
-                        }
+            drop: function (event, ui) {                
+
```
Comment (TanyaM) [inline apps/portal/js/designer.js]: New line

File: apps/portal/js/designer.js [hunk 5]
```diff
@@ -2020,7 +1977,21 @@ $(function () {
                 resize: {
                     enabled: true,
                     max_size: [12, 12],
-                    stop: function() {
+                    start: function(e, ui, widget) {
```
Comment (TanyaM) [inline apps/portal/js/designer.js]: what is e for?

File: apps/portal/js/ues-dashboards.js [hunk 6]
```diff
@@ -109,46 +141,75 @@
...
+            }
+            
+            if (container.attr('data-banner') == 'true') {
+                container.find('.ues-component-body').addClass('ues-banner-placeholder');
+            }
+            
+            if (hasComponent) {
+                createComponent(container, content[id][0], function (err) {
+                    if (err) {
+                        console.error(err);
```
Comment (TanyaM) [inline apps/portal/js/ues-dashboards.js]: Remove console.error

File: apps/portal/js/dashboard.js (hunk 1 above)
Comment (lasanthaS) [inline apps/portal/js/dashboard.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

File: apps/portal/js/dashboard.js (hunk 2 above)
Comment (lasanthaS) [inline apps/portal/js/dashboard.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

File: apps/portal/js/designer.js (hunk 3 above)
Comment (lasanthaS) [inline apps/portal/js/designer.js]: Removed the console.error. Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

File: apps/portal/js/designer.js (hunk 4 above)
Comment (lasanthaS) [inline apps/portal/js/designer.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

File: apps/portal/js/ues-dashboards.js (hunk 6 above)
Comment (lasanthaS) [inline apps/portal/js/ues-dashboards.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

**Pull request merged**